website/source/docs/guides/agent-encryption.html.md

```diff
@@ -0,0 +1,208 @@
+  "node_name": "bulldog",
+  "server": true,
+  "encrypt": "JY34uTPZyfUE+6tinMYEVw==",
+  "encrypt_verify_incoming": false,
+  "encrypt_verify_outgoing": true
+}
+```
+
+Again, use `consul reload` or `kill -HUP <process_id>` to reload the process. 
+
+**Step 4**: The previous step, enabling verify outgoing, must be completed on all agents before continuing. Update the `encrypt_verify_incoming` setting to `true` and perform a final rolling update of the cluster. 
+
+```javascript
+{
+  "data_dir": "/opt/consul",
+  "log_level": "INFO",
+  "node_name": "bulldog",
+  "server": true,
+  "encrypt": "JY34uTPZyfUE+6tinMYEVw==",
+  "encrypt_verify_incoming": true,
+  "encrypt_verify_outgoing": true
+}
+```
+
+All the agents will now be strictly enforcing encrypted gossip.
+
+## TLS Encryption for RPC
+
+Consul supports using TLS to verify the authenticity of servers and clients. To enable TLS,
+Consul requires that all clients and servers have key pairs that are generated by a single
+Certificate Authority. The CA then signs keys for each of the agents, as in
+[this tutorial on generating both a CA and signing keys](/docs/guides/creating-certificates.html)
+using [cfssl](https://cfssl.org/).
+
+TLS can be used to verify the authenticity of the servers with [`verify_outgoing`](/docs/agent/options.html#verify_outgoing) and verify the authenticity of clients with [`verify_server_hostname`](/docs/agent/options.html#verify_server_hostname).
+
+Review the [docs for specifics](https://www.consul.io/docs/agent/encryption.html).
+
+In Consul version 0.8.4 and newer, migrating to TLS encrypted traffic on a running cluster
+is supported. 
+
+### Enable TLS: New Cluster
+
+After TLS has been configured on all the agents, you can start the agents and RPC communication will be encrypted.
+
+```javascript
+{
+  "data_dir": "/opt/consul",
+  "log_level": "INFO",
+  "node_name": "bulldog",
+  "server": true,
+  "encrypt": "JY34uTPZyfUE+6tinMYEVw==",
+  "verify_incoming": true,
+  "verify_outgoing": true,
+  "verify_server_hostname": true
+}
+```
+
+The `verify_outgoing` parameter enables agents to verify the authenticity of Consul for outgoing connections. The `verify_server_hostname` paramater requires outgoing connections to perform hostname verification. Finally, the `verify_incoming` paramater enables the servers to verify the authenticity of all incoming connections. 
+
+### Enable TLS: Existing Cluster
+
+Enabling TLS on an existing cluster is supported, however this process assumes a starting point with no TLS settings configured, and involves an intermediate step in order to get to full TLS encryption.
+
+**Step 1**: Generate the necessary keys and certificate, then set the `ca_file`,`ca_path`, `cert_file`, and `key_file` settings in the configuration for each agent. Make sure the `verify_outgoing` and `verify_incoming` options are set to `false`. HTTPS for the API can be enabled at this point by setting the [`https`](/docs/agent/options.html#http_port) port.
```

Comment:
I think either `ca_file` or `ca_path` is required. The latter is just like having multiple `ca_file` entries (except you can't actually do it that way, hence the alternate setting).

Reply:
That makes sense, I've updated the sentence to include `or`.